relay/relay.go

```diff
@@ -99,3 +99,15 @@ type Peer struct {
 	// to, which is also useful for stats.
 	Zone string
 }
+
+// RateLimitDropError is an error that a Relayer will get when it should drop
+// requests due to hitting a rate limit. This is kind of a hack, because rate
+// limiting of this nature isn't part of the actual TChannel protocol.
+//
+// When the Relayer gets an error like this when calling .Get() on an outbound
+// it will record that it has dropped the packet, but *won't* notify the client.
+type RateLimitDropError struct{}
```

Comment:
There are multiple reasons for dropping traffic:
 - We are over the pool limit and are dropping traffic from the table
 - We are over some global limit for Muttley and are dropping traffic
 - Similarly, based on runtime stats, we determine that we want to drop traffic

Having a single error means that if we wanted to add more errors in future, we're stuck breaking the public API. With an interface, Muttley could return different errors for the above cases, but still have them be dropped.